aiida/engine/processes/workchains/restart.py

```diff
@@ -101,50 +112,73 @@ def handle_problem(self, node):
     `inspect_process`. Refer to their respective documentation for details.
     """
 
-    _process_class = None
+    _process_class: Optional[Type['Process']] = None
     _considered_handlers_extra = 'considered_handlers'
 
+    @property
+    def process_class(self) -> Type['Process']:
+        """Return the process class to run in the loop."""
+        from ..process import Process  # pylint: disable=cyclic-import
+        if self._process_class is None or not issubclass(self._process_class, Process):
+            raise ValueError('no valid Process class defined for `_process_class` attribute')
+        return self._process_class
+
     @classmethod
-    def define(cls, spec):
+    def define(cls, spec: 'ProcessSpec') -> None:  # type: ignore[override]
         """Define the process specification."""
-        # yapf: disable
         super().define(spec)
-        spec.input('max_iterations', valid_type=orm.Int, default=lambda: orm.Int(5),
-            help='Maximum number of iterations the work chain will restart the process to finish successfully.')
-        spec.input('clean_workdir', valid_type=orm.Bool, default=lambda: orm.Bool(False),
-            help='If `True`, work directories of all called calculation jobs will be cleaned at the end of execution.')
-        spec.input('handler_overrides',
-            valid_type=orm.Dict, required=False, validator=functools.partial(validate_handler_overrides, cls),
+        spec.input(
```

Comment:
It is good that we apply yapf too as much of the file as possible, but I would reinstate the original formatting for the `define` method here and simply add `#yapf: enable` at the end. The goal of the formatting is too make things more readable and frankly I find the formatting by yapf grotesque here. Everything is different and makes it difficult to read I find. Would it be ok to revert the formatting for the just the `define` method here?

Reply:
done (24140ade25bc5add3fe3f34dadb5d4e609b09e30)

Reply:
Cheers, much appreciated. Would you please do the same for the `CalcJob` spec definition?

Reply:
done